### PR TITLE
Start CONTRIBUTING.md with how to finish a PR

### PR DESCRIPTION
**Pull request by advayDev1**
Using PR #46 as an example.

@confile, I'd appreciate your comments on anything to clarify. It's been a while since I've done this for the first time so I may have elided steps that are second nature to me.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+## Style guide
+TODO
+
+## Testing your code
```
Comment (brunobowden) [inline CONTRIBUTING.md]: There's a lot we need to rework here

Comment (advayDev1) [inline CONTRIBUTING.md]: Definitely, I think we should do those as separate commits though - just laying out some TODOs for what we'll need in the future.   Determining a testing strategy for this project will be more involved than a guide on how to merge, while the latter could be immediately helpful.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+Now you can work on merging in master to your branch.  We'll assume a simple
+branch history here, where patch-1 first diverged from master at commit
+@5665f0a5ff774005ce30e088ab2e1b76caca5a43.
+```
+git checkout patch-1
+# condense this down to one commit to preserve proper project history
+git rebase -i 5665f0a5ff774005ce30e088ab2e1b76caca5a43
+# within the rebase editor, replace the word 'pick' with 'fixup' on
```
Comment (confile) [inline CONTRIBUTING.md]: You should claryfy where you get this number ``5665f0a5ff774005ce30e088ab2e1b76caca5a43`` from. I know it defines the request but write if it is the first or the last. 

Also: 
'pick' and  'fixup' I was totally lost here. What does this step do?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+best position to resolve these correctly.  Even if the merge is a fast-forward,
+doing the merge yourself allows you to test your code after incorporating others'
+changes.
+
+If you are new to github, here's an example workflow.
+We'll assume you are working from your local repository
+which is a clone of a fork of brunobowden/j2objc-gradle, that your feature
+branch is called 'patch-1' and that your pull request is number 46.
+
+```
```
Comment (confile) [inline CONTRIBUTING.md]: Say that the user has to checkout his cloned repo and the correct branch with.

```
git checkout --track origin/branch_branch
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# have a clean working directory and index
+# from the tip of your patch-1 branch: first save away your work
+git branch backup46
+# now don't checkout or touch branch backup46 from here on out.
+# if things go terribly wrong below, you can then return to a state of LKG sanity with:
+#git rebase --abort
+#git merge --abort
+#git checkout patch-1
+#git reset --hard backup46
+# this will return your patch-1 branch to the state before merging & rebasing
```
Comment (brunobowden) [inline CONTRIBUTING.md]: Convert lines 30-35 to a numbered list, more markdown in nature.

File: CONTRIBUTING.md (hunk 2 above)
Comment (advayDev1) [inline CONTRIBUTING.md]: good points.
Do you believe https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history is a good explanation?  If so, that is the reference I will give.  (I think best to link to experts here, no need to repeat how git works in the file).

Comment (confile) [inline CONTRIBUTING.md]: yes you can add this!

Comment (advayDev1) [inline CONTRIBUTING.md]: ptal

File: CONTRIBUTING.md (hunk 4 above)
Comment (advayDev1) [inline CONTRIBUTING.md]: ptal

**Comment by confile** [comment]
May be could could add a link how to configure the default git editor that helped me a lot.

**Comment by brunobowden** [comment]
Merged

@confile - feel free to discuss with @advayDev1 about a PR for your other suggestions

**Pull request merged**